src/code/audio/code_8019AF00.c

```diff
@@ -3503,23 +3490,452 @@ f32 sSfxOriginalPos[] = { 0.0f, 0.0f, 0.0f };
+                    } else {
+                        zVolume = 127;
+                    }
+                }
+                zVolume = sZVolumes[zVolume];
+            }
+
+            if (sAudioBaseFilter != 0) {
+                if ((bankId == BANK_PLAYER) || (bankId == BANK_ITEM) ||
+                    ((bankId == BANK_VOICE) && (((entry->sfxId & 0xFF) < 0x40) || ((entry->sfxId & 0xFF) >= 0x80)))) {
+                    baseFilter = sAudioBaseFilter;
+                    if (sAudioExtraFilter) {
+                        reverb >>= 2;
+                    }
+                }
+            }
+
+            if (((*entry->reverbAdd & 0x80) | baseFilter | sAudioExtraFilter |
+                 (entry->sfxFlags & SFX_FLAG2_APPLY_LOWPASS_FILTER)) != 0) {
+                filter = ((((*entry->reverbAdd & 0x80) >> 2) | baseFilter | sAudioExtraFilter |
+                           ((entry->sfxFlags & SFX_FLAG2_APPLY_LOWPASS_FILTER) >> 2)) *
+                          2);
+                filter &= 0xFF;
+            } else if ((sSoundMode == SOUNDMODE_SURROUND) && !(entry->sfxParams & SFX_FLAG_VOLUME_NO_DIST)) {
+                filter = AudioSfx_ComputeSurroundSoundFilter(behindScreenZ, entry, panSigned);
+            }
+            break;
+
+        case BANK_SYSTEM:
+            break;
+    }
+
+    if (sSfxChannelState[channelIndex].volume != volume) {
+        volumeS8 = (u8)(volume * 127.0f);
+        sSfxChannelState[channelIndex].volume = volume;
+    } else {
+        volumeS8 = -1;
+    }
+
+    //! FAKE:
+    if (1) {}
+
+    // CHAN_UPD_SCRIPT_IO (slot 2, sets volume)
+    AudioThread_QueueCmdS8((0x6 << 24) | (SEQ_PLAYER_SFX << 16) | (channelIndex << 8) | 2, volumeS8);
+
+    if (sSfxChannelState[channelIndex].reverb != reverb) {
+        sSfxChannelState[channelIndex].reverb = reverb;
+    }
+    if (sSfxChannelState[channelIndex].freqScale != freqScale) {
+        sSfxChannelState[channelIndex].freqScale = freqScale;
+    }
+    if (sSfxChannelState[channelIndex].stereoBits != 0) {
+        sSfxChannelState[channelIndex].stereoBits = 0;
+    }
+    if (sSfxChannelState[channelIndex].filter != filter) {
+        sSfxChannelState[channelIndex].filter = filter;
+    }
+    if (sSfxChannelState[channelIndex].combFilterGain != combFilterGain) {
+        sSfxChannelState[channelIndex].combFilterGain = combFilterGain;
+    }
+    if (sSfxChannelState[channelIndex].zVolume != zVolume) {
+        sSfxChannelState[channelIndex].zVolume = zVolume;
+    }
+    if (sSfxChannelState[channelIndex].panSigned != panSigned) {
+        sSfxChannelState[channelIndex].panSigned = panSigned;
+    }
+}
+
+u32 AudioSfx_SetFreqAndStereoBits(u8 seqScriptValIn, SequenceChannel* channel) {
+    channel->stereo.asByte = sSfxChannelState[seqScriptValIn].stereoBits;
+    channel->freqScale = sSfxChannelState[seqScriptValIn].freqScale;
+    channel->changes.s.freqScale = true;
+
+    return seqScriptValIn;
+}
+
+void AudioSfx_ResetSfxChannelState(void) {
```

Comment:
I've definitely been conscious of when to include `Sfx` in the function name and when not to. In this particular case, I included it because the struct itself is called `SfxChannelState`, but `AudioSfx_ResetChannelState` could work too